### ISSUE TITLE
Enable focus outline on the map element(s)

### ISSUE THREAD

**Issue by Malvoz** [Maps4HTML/Web-Map-Custom-Element#273]
Separating the relevant parts from the discussion in https://github.com/Maps4HTML/Web-Map-Custom-Element/issues/260 about enabling the focus outline for map elements:

@prushforth wrote https://github.com/Maps4HTML/Web-Map-Custom-Element/issues/260#issue-794537387:

> @Malvoz I was looking at our mapml-viewer today while tabbing, and I noticed that there is no focus outline on the map when focused. Should we fix this?

@Malvoz wrote https://github.com/Maps4HTML/Web-Map-Custom-Element/issues/260#issuecomment-768128229:
> Showing an outline on `.leaflet-container` on focus would be ideal, but Leaflet tends to do strange things with focus management in general, it produces an inconsistent or "flashing" outline when operating the zoom controls with a mouse pointer:
> 
> https://user-images.githubusercontent.com/26493779/105957381-7975ba80-6079-11eb-9620-576718196ae1.mp4
> 
> A second problem is that some content inside the map such as the attribution container and the map graphics are overlapping the outline.
> 
> >  Should we fix this?
> 
> If we can, yes. It is still a failure of SC [2.4.7 Focus Visible](https://www.w3.org/TR/WCAG21/#focus-visible) even though we've [fixed outlines](https://github.com/Maps4HTML/Web-Map-Custom-Element/blob/24245084ca66488605ad0e7ced1de621fc4d491f/src/mapml.css#L328-L335) for all the other focusable components.
> 
> Edit: One could argue that the issues described above shouldn't dissaude us from displaying a visible focus indicator. Should we just enable it?

@ahmadayubi wrote https://github.com/Maps4HTML/Web-Map-Custom-Element/issues/260#issuecomment-768357363:
> Could [focus-within](https://developer.mozilla.org/en-US/docs/Web/CSS/:focus-within) be used?

@Malvoz wrote https://github.com/Maps4HTML/Web-Map-Custom-Element/issues/260#issuecomment-768398149:

> @ahmadayubi focus is still triggered the same way, producing the same behavior as `:focus` unfortunately. I also tested the relatively new [`:focus-visible`](https://developer.mozilla.org/en-US/docs/Web/CSS/:focus-visible) which can be used to only show outlines when the user is using a keyboard, but it seems the UA heuristics determine a focus outline should be drawn anyway, supposedly due to programmatic focus.

@ahmadayubi wrote https://github.com/Maps4HTML/Web-Map-Custom-Element/issues/260#issuecomment-768405962:

> Ah okay, I recently added the ability to focus vector features so I was thinking `focus-within` could be used to outline the map while the feature was focused as additional feedback to let the user know they are tabbing through the elements within the map.

@Malvoz wrote https://github.com/Maps4HTML/Web-Map-Custom-Element/issues/260#issuecomment-768409870:

> Why isn't outline set to the vector feature directly in that case? That way the user would know which feature as focus. Is it because features outside the map view can be focused?

@ahmadayubi wrote https://github.com/Maps4HTML/Web-Map-Custom-Element/issues/260#issuecomment-768410947:

> No the feature is outlined but it might be helpful to have both the feature focused outlined and the map outlined at the same time. Similar to how google maps outlines the entire map when you are navigating with keyboard and tabs.

@prushforth wrote https://github.com/Maps4HTML/Web-Map-Custom-Element/issues/260#issuecomment-768607093:

> > One could argue that the issues described above shouldn't dissuade us from displaying a visible focus indicator. Should we just enable it?
>
> I think so. I asked Ahmad to put the cross-hair on when the map is focused. We can deal with the focus outline as a separate issue, since we may have to figure out how to make it work better than is currently possible, it seems.
